### PR TITLE
Fix short circuit functions and mutations with sparse arguments

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed execution of short circuit functions as well as `UPDATE` mutations with sparse columns.

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit 8ff5fda62816f7973466fe3a6a8923fb88ef4a1f with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49716/8ff5fda62816f7973466fe3a6a8923fb88ef4a1f/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Columns/ColumnFunction.cpp [hunk 1]
```diff
@@ -264,7 +264,9 @@ void ColumnFunction::appendArgument(const ColumnWithTypeAndName & column)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot capture column {} because it has incompatible type: "
                         "got {}, but {} is expected.", argument_types.size(), column.type->getName(), argument_types[index]->getName());
 
-    captured_columns.push_back(column);
+    auto captured_column = column;
+    captured_column.column = captured_column.column->convertToFullColumnIfSparse();
+    captured_columns.push_back(std::move(captured_column));
```
Comment (vmarora) [inline src/Columns/ColumnFunction.cpp:269]: consider using emplace_back.

**Review by alexey-milovidov — APPROVED** [review]

File: src/Columns/ColumnFunction.cpp (hunk 1 above)
Comment (CurtizJ) [reply src/Columns/ColumnFunction.cpp:269]: In that case there is not difference between `push_back` and `emplace_back`. Both will call move constructor.

Comment (vmarora) [reply src/Columns/ColumnFunction.cpp:269]: You are right. With move, push_back behavior is same.

**Pull request merged**